rdr_service/resource/generators/workbench.py

```diff
@@ -0,0 +1,265 @@
+
+            data['sex_at_birth'] = str(WorkbenchWorkspaceSexAtBirth(row.sex_at_birth))
+            data['sex_at_birth_id'] = int(WorkbenchWorkspaceSexAtBirth(row.sex_at_birth))
+
+            data['gender_identity'] = str(WorkbenchWorkspaceGenderIdentity(row.gender_identity))
+            data['gender_identity_id'] = int(WorkbenchWorkspaceGenderIdentity(row.gender_identity))
+
+            data['sexual_orientation'] = str(WorkbenchWorkspaceSexualOrientation(row.sexual_orientation))
+            data['sexual_orientation_id'] = int(WorkbenchWorkspaceSexualOrientation(row.sexual_orientation))
+
+            data['geography'] = str(WorkbenchWorkspaceGeography(row.geography))
+            data['geography_id'] = int(WorkbenchWorkspaceGeography(row.geography))
+
+            data['disability_status'] = str(WorkbenchResearcherDisability(row.disability_status))
+            data['disability_status_id'] = int(WorkbenchResearcherDisability(row.disability_status))
+
+            data['access_to_care'] = str(WorkbenchWorkspaceAccessToCare(row.access_to_care))
+            data['access_to_care_id'] = int(WorkbenchWorkspaceAccessToCare(row.access_to_care))
+
+            data['education_level'] = str(WorkbenchWorkspaceEducationLevel(row.education_level))
+            data['education_level_id'] = int(WorkbenchWorkspaceEducationLevel(row.education_level))
+
+            data['income_level'] = str(WorkbenchWorkspaceIncomeLevel(row.income_level))
+            data['income_level_id'] = int(WorkbenchWorkspaceIncomeLevel(row.income_level))
+
+            data['access_tier'] = str(WorkbenchWorkspaceAccessTier(row.access_tier))
+            data['access_tier_id'] = int(WorkbenchWorkspaceAccessTier(row.access_tier))
+
+            return generators.ResourceRecordSet(schemas.WorkbenchWorkspaceSchema, data)
+
+
+def res_workspace_batch_update(_ids):
+    """
+    Update a batch of ids.
+    :param _ids: list of ids
+    """
+    gen = generators.WBWorkspaceGenerator()
+    for _id in _ids:
+        res = gen.make_resource(_id)
+        res.save()
+
+
+class WBWorkspaceUsersGenerator(generators.BaseGenerator):
+    """
+    Generate a workbench workspace user resource object
+    """
+    ro_dao = None
+
+    def make_resource(self, pk_id, backup=False):
+        """
+        Build a resource object from the given primary key id.
+        :param pk_id: Primary key value.
+        :param backup: if True, get from backup database instead of Primary.
+        :return: resource object
+        """
+        if not self.ro_dao:
+            self.ro_dao = ResourceDataDao(backup=backup)
+
+        with self.ro_dao.session() as ro_session:
+            row = ro_session.execute(
+                text('select * from rdr.workbench_workspace_user_history where id = :id order by modified desc'),
+                {'id': pk_id}).first()
+            # Fall back to workbench_workspace_user, it looks like there are missing records in the history table.
```

Comment:
We could add a 'limit 1' to the SQL statements to ensure only one record is returned to SQLAlchemy instead of returning all matching rows and then only using the first row.  It's not big enough enough performance problem to worry about though.